program-runtime/src/loaded_programs.rs

```diff
@@ -135,9 +140,14 @@ impl solana_frozen_abi::abi_example::AbiExample for LoadedPrograms {
     }
 }
 
+pub enum LoadedProgramEntry {
+    PreExisting(Arc<LoadedProgram>),
+    Inserted(Arc<LoadedProgram>),
+}
+
 impl LoadedPrograms {
     /// Inserts a single entry
-    pub fn insert_entry(&mut self, key: Pubkey, entry: LoadedProgram) -> bool {
+    pub fn insert_entry(&mut self, key: Pubkey, entry: LoadedProgram) -> LoadedProgramEntry {
```

Comment:
I don't think this change belongs into this PR.
At least `insert_entry()` is not used yet.